src/Common/src/System/LocalAppContext.cs

```diff
@@ -40,6 +38,17 @@ private static bool GetCachedSwitchValueInternal(string switchName, ref int swit
             return isSwitchEnabled;
         }
 
+        private static void InitDisableCaching()
+        {
+            bool isEnabled;
+            if (AppContext.TryGetSwitch(@"TestSwitch.LocalAppContext.DisableCaching", out isEnabled))
```

Comment:
If you do that, then you don't have to introduce an extra check in the `GetCachedSwitchValueInternal` method as it will become part of the `DisableCaching` check.